src/graph_generator.cc

```diff
@@ -473,6 +474,27 @@ namespace detail {
 		}
 	}
 
+	void graph_generator::process_task_side_effect_requirements(const task_id tid) {
+		const auto tsk = task_mngr.get_task(tid);
+		for(const auto cmd : cdag.task_commands(tid)) {
```

Comment:
Maybe skip this loop / early return from the function if the side effect map is empty? Probably doesn't matter much, but it's easy to do, likely the default case, and might have some impact with large clusters (and thus large numbers of task commands).

Reply:
Done.